src/wp-includes/media.php

```diff
@@ -5444,25 +5444,38 @@ function wp_get_webp_info( $filename ) {
  *                     that the `loading` attribute should be skipped.
  */
 function wp_get_loading_attr_default( $context ) {
-	// Only elements with 'the_content' or 'the_post_thumbnail' context have special handling.
-	if ( 'the_content' !== $context && 'the_post_thumbnail' !== $context ) {
-		return 'lazy';
+	// Skip lazy-loading for the overall block template, as it is handled more granularly.
+	if ( 'template' === $context ) {
+		return false;
 	}
 
-	// Only elements within the main query loop have special handling.
-	if ( is_admin() || ! in_the_loop() || ! is_main_query() ) {
-		return 'lazy';
+	// Do not lazy-load images in the header block template part, as they are likely above the fold.
+	$header_area = WP_TEMPLATE_PART_AREA_HEADER;
+	if ( "template_part_{$header_area}" === $context ) {
+		return false;
 	}
 
-	// Increase the counter since this is a main query content element.
-	$content_media_count = wp_increase_content_media_count();
+	// The first elements in 'the_content' or 'the_post_thumbnail' should not be lazy-loaded,
+	// as they are likely above the fold.
```

Comment:
```suggestion
	/*
	 * The first elements in 'the_content' or 'the_post_thumbnail' should not be lazy-loaded,
	 * as they are likely above the fold.
	 */
```
Just a slight adjustment to use the [multiline comment format](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#5-2-multi-line-comments) here.